src/style-spec/function/expressions.js

```diff
@@ -0,0 +1,425 @@
+'use strict';
+
+// @flow
+
+const assert = require('assert');
+
+const {
+    NumberType,
+    StringType,
+    BooleanType,
+    ColorType,
+    ObjectType,
+    ValueType,
+    InterpolationType,
+    typename,
+    variant,
+    vector,
+    array,
+    anyArray,
+    lambda,
+    nargs
+} = require('./types');
+
+/*::
+ import type { PrimitiveType, TypeName, VariantType, VectorType, ArrayType, AnyArrayType, NArgs, LambdaType, Type } from './types.js';
+
+ import type { TypeError, TypedLambdaExpression, TypedLiteralExpression, TypedExpression } from './type_check.js';
+
+ import type { CompiledExpression } from './compile.js';
+
+ import type { ExpressionName } from './expression_name.js';
+
+ export type Definition = {
+     name: ExpressionName,
+     type: LambdaType,
+     compile: (args: Array<CompiledExpression>, expr: TypedLambdaExpression) => ({ js?: string, errors?: Array<string>, isFeatureConstant?: boolean, isZoomConstant?: boolean })
+ }
+ */
+
+const expressions: { [string]: Definition } = {
+    'ln2': defineMathConstant('ln2'),
+    'pi': defineMathConstant('pi'),
+    'e': defineMathConstant('e'),
+    'string': {
+        name: 'string',
+        type: lambda(StringType, ValueType),
+        compile: args => ({ js: `String(${args[0].js})` })
```

Comment:
Is it wise to inherit all of JS's coercion quirks?

* `["string", ["^", 2, 70]]` ⇢ "1.1805916207174113e+21"
* `["string", ["properties"]]` ⇢ "[object Object]"
* `["string", ["array", 1, 2]]` ⇢ "1,2"
* `["string, ["get", "weird"]]` + `{"properties": {"weird": {"toString": "asdf"}}` ⇢ TypeError